src/axom/core/execution/internal/omp_exec.hpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2017-2019, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef AXOM_OMP_EXEC_HPP_
+#define AXOM_OMP_EXEC_HPP_
+
+#include "axom/config.hpp"
+#include "axom/core/memory_management.hpp"
+
+// RAJA includes
+#include "RAJA/RAJA.hpp"
+
+#ifndef RAJA_ENABLE_OPENMP
+#error *** OMP_EXEC requires an OpenMP enabled RAJA ***
+#endif
+
+// Umpire includes
+#ifdef AXOM_USE_UMPIRE
+#include "umpire/Umpire.hpp"
+#endif
+
+namespace axom
+{
+
+/*!
+ * \brief Indicates parallel execution on the CPU using OpenMP.
+ */
+struct OMP_EXEC{ };
+
+/*!
+ * \brief execution_space traits specialization for OMP_EXEC
+ */
+template < >
+struct execution_space< OMP_EXEC >
+{
+  using loop_policy = RAJA::omp_parallel_for_exec;
+
+  /* *INDENT-OFF* */
+  using loop2d_policy = RAJA::KernelPolicy<
```

Comment:
For most kernels, using OpenMP collapse on all loop levels performs poorer than applying OpenMP to a subset.